src/codegen/doc/mod.rs

```diff
@@ -503,7 +513,30 @@ where
             )?;
         }
 
-        for parameter in fn_.parameters() {
+        let parameters: Vec<_> = fn_
+            .parameters()
+            .iter()
+            .filter(|param| {
+                let is_array_length_param = if param.name.len() > 3 {
+                    let array_param_name = &param.name[2..];
+                    &param.name[0..2] == "n_"
+                        && fn_
+                            .parameters()
+                            .iter()
+                            .find(|p| p.name == array_param_name)
+                            .is_some()
+                } else {
+                    false
+                };
```

Comment:
this seems to work pretty fine at least on gtk4-rs, there are three types where that convention fails: ListStore, TreeModelFilter & TreeStore. Imho, i would say it's worth it (of course, in such case the unnecessary params are kept)  and the correct solution here would be to rename the params upstream so that we can use such convention without worrying too much.